tools/profiles/minimal-printf.json

```diff
@@ -0,0 +1,20 @@
+{
+    "GCC_ARM": {
+       "common": ["-DMBED_MINIMAL_PRINTF", "-fno-builtin-printf"],
+        "ld": ["-Wl,--wrap,printf", "-Wl,--wrap,sprintf", "-Wl,--wrap,snprintf",
+               "-Wl,--wrap,vprintf", "-Wl,--wrap,vsprintf", "-Wl,--wrap,vsnprintf",
+               "-Wl,--wrap,fprintf", "-Wl,--wrap,vfprintf"]
+    },
+    "ARMC6": {
+        "common": ["-DMBED_MINIMAL_PRINTF", "-fno-builtin-printf"]
+    },
+    "ARM": {
+        "common": ["-DMBED_MINIMAL_PRINTF"]
+    },
+    "uARM": {
```

Comment:
Should we document somewhere that the user needs to remove `"-fno-builtin-printf"` if s/he decides to use Arm Compiler 5 with `uARM`? It is not supported and will generate an error.
As a reminder, the user is able to select ARMC5 for uARM by adding `"target.supported_toolchains": ["ARMC5", "uARM"]` in `mbed_app.json`.

Reply:
To solve this issue `-fno-builtin-printf"` should be removed for `uARM`

Reply:
Commit  https://github.com/ARMmbed/mbed-os/pull/11051/commits/d6b197a98bbfd84b6a981aece6b39445399c24fc

By default, when building with `-t uARM` profile, ARMC6 compiler is used (as expected) and the `"ARMC6"` build profile flags are used.

In order to build `-t uARM` with ARMC5 compiler, I added
``` 
"target.supported_toolchains": ["ARMC5", "uARM"]
```
in `mbed_app.json`. In this case ARMC5 compiler is used (as expected) and the `"uARM"` build profile flags are used.
For info, I noticed that the `"uARM"` compilation flags are a superset of `"ARM"` compilation flags.

In summary, I think it is correct to remove `no-builtin-printf` from the list of uARM compilation flags but I don't know how the compilation flags for uARM (for example -DMBED_RTOS_SINGLE_THREAD -D__MICROLIB) are added to ARMC6 compilation flags in the default case.

@madchutney @theotherjimmy